client/src/pages/Signup.js

```diff
@@ -0,0 +1,220 @@
+import { withStyles } from '@material-ui/core/styles';
+import * as imageURL from '../images/login-photo.png'
+import { Link } from 'react-router-dom'
+import Snackbar from '@material-ui/core/Snackbar'
+import SnackbarContent from '@material-ui/core/SnackbarContent'
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    '& .MuiTextField-root': {
+      margin: theme.spacing(1),
+      width: '25ch',
+      length: '400ch',
+      display: 'flex',
+      flexDirection: 'column',
+      marginBottom: '30px'
+    }
+  },
+}));
+
+const ContinueButton = withStyles({
+  root: {
+    background: 'linear-gradient(45deg, rgba(73, 94, 203, .8) 30%, rgba(73, 145, 203, .8) 90%)',
+    color: 'white',
+    height: 48,
+    padding: '0 30px',
+    boxShadow: '0 3px 5px 2px rgba(125, 123, 135, .3)',
+    marginLeft: '10px'
+  },
+})(Button);
+
+const LoginButton = withStyles({
+  root: {
+    background: 'linear-gradient(45deg, rgba(105, 105, 105, .8) 30%, rgba(169, 169, 169, .8) 90%)',
+    color: 'white',
+    height: '5ch',
+    padding: '0 30px',
+    boxShadow: '0 3px 5px 2px rgba(125, 123, 135, .3)',
+    marginLeft: '10px'
+  },
+})(Button);
+
+const styles = {
+  root: {
+    background: 'red'
+  }
+};
+
+const Signup = () => {
+
+  const classes = useStyles();
+
+  const [formData, setFormData] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const [state, setState] = useState({
+    open: false,
+    vertical: 'bottom',
+    horizontal: 'center',
+    message: null
+  });
+
+  const { vertical, horizontal, open, message } = state;
+
+  const { firstName, lastName, email, password, confirmPassword } = formData;
+
+  const showAlert = (newState) => {
+    setState({ open: true, ...newState });
+  }
+
+  const handleClose = () => {
+    setState({ ...state, open: false });
+  }
+
+  const onChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  }
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    // login(email, password);
+
+    if (!areAllFieldsCompleted()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Must fill out every field!" })
+      return
+    }
+    if (!passwordIsValid() && !emailIsValid()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Password & E-mail are invalid!" })
+      return
+    }
+    if (!passwordIsValid() && password.length !== 0) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Password is invalid!" })
+      return
+    }
+    if (!emailIsValid() && email.length !== 0) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "E-mail is invalid!" })
+      return
+    }
+    if (areAllFieldsCompleted() && emailIsValid && !passwordsAreTheSame()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Passwords must match!" })
+      return
```

Comment:
Since all alerts are shown bottom and center, maybe you don't need to specify those parameters all the time, and showAlert just takes in a message